### ISSUE TITLE
Intensive testing in wazuh-logcollector

### ISSUE THREAD

**Issue by JcabreraC** [wazuh/wazuh#8484]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| - | wazuh-logcollector | Agent | - | macOS Sierra or greater |

Hi Team!

As part of the epic https://github.com/wazuh/wazuh/issues/4572, it is necessary to carry out intensive tests of the new development in charge of reading logs in MacOS (ULS), as there are currently no logcollector integration tests.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux Debug
  - [x] Linux
  - [x] Windows Debug
  - [x] MAC OS X Debug
  - [x] Windows
  - [x] MAC OS X
  - [x] FreeBSD
  - [x] FreeBSD Debug
- [ ] Source installation
- [ ] Package installation
- [ ] Source upgrade
- [ ] Package upgrade
- [ ] Review logs syntax and correct language
- [ ] QA templates contemplate the added capabilities

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Coverity
  - [x] Valgrind (memcheck and descriptor leaks check)
  - [ ] Dr. Memory
  - [x] AddressSanitizer
- Memory tests for Windows
  - [x] Scan-build report
  - [ ] Coverity
  - [ ] Dr. Memory
- Memory tests for macOS
  - [ ] Scan-build report
  - [x] Coverity (Launched for linux agent, but without `darwin` guards)
  - [ ] Leaks
  - [ ] AddressSanitizer
  - [ ] ThreadSanitizer

- [ ] Retrocompatibility with older Wazuh versions
- [ ] Working on cluster environments
- [ ] Configuration on demand reports new parameters
- [ ] The data flow works as expected (agent-manager-api-app)
- [ ] Added unit tests (for new features)
- [ ] Stress test for affected components

### Configuration
- [x] Only one localfile block with `macos` as `log_format` is allowed
- [x] If log_format is `macos` then location should be `macos` as well
- [x] Query’s `type` attribute is correctly handled
- [x] Query’s `level` attribute is correctly handled
- [x] If log_format is `macos`: `only-future-events`, `target`, `out_format` attributes are allowed
- [x] If log_format is `macos`: `command`, `alias`, `ignore_binaries`, `age`, `exclude`, `reconnect_time` and `label` are not allowed and logcollector execution must continue. 

### Runtime
- [x] `log stream` and `log show` processes are closed after logcollector exits

Regards.

**Comment by jnasselle** [comment]
### `out_format` validation
macOS agent localfile
```
 <localfile>
  <log_format>macos</log_format>
  <location>macos</location>
  <query type="activity"> process == 'AirPlayXPCHelper' </query>
  <out_format>Test $(timestamp %Y-%m-%d %H:%M:%S): $(log)</out_format>
</localfile>
```
Manager's `archives.log`
```json
{"timestamp":"2021-05-12T11:59:53.777+0000","agent":{"id":"002","name":"macos-1100","ip":"10.0.2.15"},"manager":{"name":"wazuh-dev"},"id":"1620820793.538739","full_log":"Test 2021-05-12 04:59:48: 2021-05-12 04:59:43.772231-0700  localhost AirPlayXPCHelper[131]: (CoreUtils) Created Activity ID: 0x9be9b, Description: CoreUtils/CUPairingManager/getPairedPeersWithOptions","decoder":{},"location":"macos"}
{"timestamp":"2021-05-12T11:59:55.787+0000","agent":{"id":"002","name":"macos-1100","ip":"10.0.2.15"},"manager":{"name":"wazuh-dev"},"id":"1620820795.538739","full_log":"Test 2021-05-12 04:59:50: 2021-05-12 04:59:46.796446-0700  localhost AirPlayXPCHelper[131]: (CoreUtils) Created Activity ID: 0x9be9c, Description: CoreUtils/CUPairingManager/getPairedPeersWithOptions","decoder":{},"location":"macos"}
```

**Comment by jnasselle** [comment]
### Invalid options
macOS agent localfile
```xml
<localfile>
  <log_format>macos</log_format>
  <location>macos</location>
  <query type="activity"> process == 'AirPlayXPCHelper' </query>
  <label key="@source">myapp</label>
  <frequency>12</frequency>
  <alias>testing</alias>
  <ignore_binaries>yes</ignore_binaries>
  <age>1</age>
  <exclude>test</exclude>
  <multiline_regex replace="wspace">^Traceback</multiline_regex>
  <reconnect_time>10s</reconnect_time>
  <only-future-events>no</only-future-events>
</localfile>
```
macOS agent `ossec.log`
```
2021/05/12 05:29:43 wazuh-logcollector[68497] localfile-config.c:497 at Read_Localfile(): WARNING: (8004): log_format 'macos' does not support 'reconnect_time' option. Option will be ignored.
2021/05/12 05:29:43 wazuh-logcollector[68497] localfile-config.c:500 at Read_Localfile(): WARNING: (8004): log_format 'macos' does not support 'age' option. Option will be ignored.
2021/05/12 05:29:43 wazuh-logcollector[68497] localfile-config.c:503 at Read_Localfile(): WARNING: (8004): log_format 'macos' does not support 'ignore_binaries' option. Option will be ignored.
2021/05/12 05:29:43 wazuh-logcollector[68497] localfile-config.c:506 at Read_Localfile(): WARNING: (8004): log_format 'macos' does not support 'exclude' option. Option will be ignored.
2021/05/12 05:29:43 wazuh-logcollector[68497] localfile-config.c:509 at Read_Localfile(): WARNING: (8004): log_format 'macos' does not support 'multiline_regex' option. Option will be ignored.
2021/05/12 05:29:43 wazuh-logcollector[68497] localfile-config.c:512 at Read_Localfile(): WARNING: (8004): log_format 'macos' does not support 'label' option. Option will be ignored.
2021/05/12 05:29:43 wazuh-logcollector[68497] localfile-config.c:515 at Read_Localfile(): WARNING: (8004): log_format 'macos' does not support 'frequency' option. Option will be ignored.
2021/05/12 05:29:43 wazuh-logcollector[68497] localfile-config.c:518 at Read_Localfile(): WARNING: (8004): log_format 'macos' does not support 'alias' option. Option will be ignored.
2021/05/12 05:29:43 wazuh-logcollector[68497] localfile-config.c:539 at Read_Localfile(): WARNING: (8001): log_format 'macos' does not support multiline_regex option. Will be ignored.
```

**Comment by jnasselle** [comment]
### Coverity
Two incidences solved in #8636 
![image](https://user-images.githubusercontent.com/1791430/118053313-aadd7200-b35a-11eb-8610-8740a9cc22c3.png)
